api/v1beta1/placementapi_types.go

```diff
@@ -86,6 +86,11 @@ type PlacementAPISpec struct {
 	// actual action pod gets started with sleep infinity
 	Debug PlacementAPIDebug `json:"debug,omitempty"`
 
+	// +kubebuilder:validation:Optional
+	// PlacementAPIImmutable - make some specs immutable to prohibit update operations
+	// not fully supported by this operator.
+	Immutable PlacementAPIImmutable `json:"immutable,omitempty"`
```

Comment:
I have a bad feeling about the UX of having a separate struct / map defining what field is settable and when. But I have no good suggestion how to force an extra round-trip for the user to make sure the DB change is intentional.

Maybe(?) we can have a single "advanced-mode" flag for the whole CRD. That is defaulted to false, and prevents any dangerous operation. That would allow us not to duplicate fields directly.

Reply:
This is addressed in the latest version.